packages/rest/src/parser.ts

```diff
@@ -104,19 +110,19 @@ function buildOperationArguments(
     const spec = paramSpec as ParameterObject;
     switch (spec.in) {
       case 'query':
-        args.push(request.query[spec.name]);
+        args.push(deserialize(request.query[spec.name], spec));
         break;
       case 'path':
-        args.push(pathParams[spec.name]);
+        args.push(deserialize(pathParams[spec.name], spec));
         break;
       case 'header':
-        args.push(request.headers[spec.name.toLowerCase()]);
+        args.push(deserialize(request.headers[spec.name.toLowerCase()], spec));
         break;
       case 'formData':
-        args.push(body ? body[spec.name] : undefined);
+        args.push(body ? deserialize(body[spec.name], spec) : undefined);
         break;
       case 'body':
-        args.push(body);
+        args.push(deserialize(body, spec));
```

Comment:
To my taste, this code is repeating the call of `deserialize` too much.

Please rework the code so that the `switch` block reads the argument value from the correct place in the request (query, pathParams, etc.) and then move calls of `deserialize` and `args.push` to a single place after the `switch` block.